common/src/main/java/io/netty/util/ResourceLeakDetector.java

```diff
@@ -333,12 +332,52 @@ protected void reportUntracedLeak(String resourceType) {
     protected void reportInstancesLeak(String resourceType) {
     }
 
+    /**
+     * This is an enhanced version of ResourceLeak.
+     * It will not clean up records when it is closed, because we may still use records in the future.
+     */
+    private static final class AdvancedResourceLeak<T> extends DefaultResourceLeak<T> {
+
+        AdvancedResourceLeak(
+                Object referent,
+                ReferenceQueue<Object> refQueue,
+                Set<DefaultResourceLeak<?>> allLeaks) {
+            super(referent, refQueue, allLeaks);
+        }
+
+        @Override
+        public boolean close() {
+            if (allLeaks.remove(this)) {
+                // Call clear so the reference is not even enqueued.
+                clear();
+                // We didn't use `headUpdater.set(this, null)` to clear head immediately.
+                return true;
+            }
+            return false;
+        }
+
+        private void clearRecord() {
+            if (headUpdater.get(this) != null) {
+                headUpdater.set(this, null);
+            }
+        }
+
+        @Override
+        protected void finalize() throws Throwable {
```

Comment:
Do we need release head when `AdvancedResourceLeak`  will be gced..?

Reply:
IMHO this finalize should not be here.